sample/helloworld/test.py

```diff
@@ -0,0 +1,3 @@
+def harness(p):
+    p.expect_exact('Hello world!')
```

Comment:
this is not the best idea, because to detect FAIL we need to wait 30s.

Please provide some more sensible examples (maybe using `readline()` for this and similar tests is enough?)

Maybe all harnesses should detect `psh >` and conclude the program has ended/failed with return code? (should we detect the error code?)

Reply:
Now after invoking `exec prog` shell won't return to us (or I don't know how to do it). So is impossible to detect `psh >` after program execution.

Reply:
uch, I've always used `/bin/helloworld` which spawns new process, so after it's finish we're getting back to `psh`. I haven't checked the psh `exec` implementation, but I believe it uses `execve`, so swaps the current process contents with new one.

I think spawning new processes for "big" targets is a good execution strategy.

Reply:
I added spawning new process. Parsing `psh >` after execution looks like a nice idea for additional check. I've written it down in my todo list for next PR.